lib/types/btrfs.nix

```diff
@@ -126,8 +126,10 @@ in
             MNTPOINT=$(mktemp -d)
             mount ${config.device} "$MNTPOINT" -o subvol=/
             trap 'umount $MNTPOINT; rm -rf $MNTPOINT' EXIT
-            btrfs subvolume create "$MNTPOINT"/${subvol.name} ${toString subvol.extraArgs}
-            ${swapCreate "$MNTPOINT/${subvol.name}" subvol.swap}
+            SUBVOL_ABS_PATH="$MNTPOINT/${subvol.name}"
+            mkdir -p '${lib.dirOf "$SUBVOL_ABS_PATH"}'
```

Comment:
I don't think this can work nix. Cannot expand this bash variable here. Can we not just use `dirname` here?

Reply:
You're right. It appeared to work in my quick vm test, but I had forgotten to update the lock file prior to running it. Apologies.

The PR is now updated to use `dirname`.